lib/internal/worker.js

```diff
@@ -98,6 +98,9 @@ class Worker extends EventEmitter {
       argv = options.argv.map(String);
     }
     if (!options.eval) {
+      if (filename.toString().startsWith('file://')) {
```

Comment:
Given that calling `toString()` on a `URL` object is a relatively expensive operation, I would structure this check a bit differently... Something like:

```js
if ((filename instanceof URL && filename.protocol === 'file:') ||
    filename.toString().startsWith('file://')) {
```

Reply:
The current code doesn't accept `URL` objects because of the `validateString(filename, 'filename');` line earlier.

Reply:
Ah right, so either that needs to be fixed or the doc change should not list `URL` as an option